MSAL/src/MSALDeviceInformation.m

```diff
@@ -32,20 +32,21 @@
 #import "ASAuthorizationSingleSignOnProvider+MSIDExtensions.h"
 
 NSString *const MSAL_DEVICE_INFORMATION_SSO_EXTENSION_FULL_MODE_KEY = @"isSSOExtensionInFullMode";
+NSString *const MSAL_DEVICE_INFORMATION_UPN_ID_KEY = @"upnIdentifier";
+NSString *const MSAL_DEVICE_INFORMATION_AAD_DEVICE_ID_KEY = @"aadDeviceIdentifier";
+NSString *const MSAL_DEVICE_INFORMATION_AAD_TENANT_ID_KEY = @"aadTenantIdentifier";
 
 @implementation MSALDeviceInformation
 {
     NSMutableDictionary *_extraDeviceInformation;
 }
 
-- (instancetype)initWithMSIDDeviceInfo:(MSIDDeviceInfo *)deviceInfo
+- (instancetype)init
```

Comment:
I think device info should still be needed, no?

Reply:
Yes deviceInfo is still required, its just I moved it from constructor to setter method.
So now we can initialise object without MSIDDeviceInfo and that helps up to add WPJ info and of DeviceInfo is available then just add using addExtraDeviceInformation() method.